src/lib.rs

```diff
@@ -39,19 +39,62 @@
 //! # }
 //! ```
 
+#[cfg(feature = "serde")]
+extern crate serde;
+#[cfg(feature = "serde")]
+use serde::{Deserialize, Serialize};
+
 use std::borrow::Borrow;
 use std::collections::hash_map;
 use std::collections::HashMap;
 use std::fmt::{self, Debug};
 use std::hash::Hash;
 
+#[cfg(feature = "serde")]
+#[derive(Eq, Serialize, Deserialize)]
+#[serde(from = "HashMap<K1, (K2, V)>")]
```

Comment:
Can we use `cfg_attr` to avoid duplicating the `struct` definition? How do other `serde` using crates handle this?